bin/faucet/src/main.rs

```diff
@@ -229,3 +238,122 @@ fn long_version() -> LongVersion {
         debug: option_env!("VERGEN_CARGO_DEBUG").unwrap_or_default(),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::{
+        env::temp_dir,
+        io::{BufRead, BufReader},
+        path::PathBuf,
+        process::{Command, Stdio},
+        str::FromStr,
+    };
+
+    use fantoccini::ClientBuilder;
+    use serde_json::{json, Map};
+    use url::Url;
+
+    use crate::{config::FaucetConfig, run_faucet_command, stub_rpc_api::serve_stub, Cli};
+
+    /// This test starts a stub node, a faucet connected to the stub node, and a chromedriver
+    /// to test the faucet website. It then loads the website and checks that all the requests
+    /// made return status 200.
+    #[tokio::test]
+    async fn test_website() {
+        let stub_node_url = Url::from_str("http://localhost:50051").unwrap();
+
+        // Start the stub node
+        tokio::spawn({
+            let stub_node_url = stub_node_url.clone();
+            async move { serve_stub(&stub_node_url).await.unwrap() }
+        });
+
+        let config_path = temp_dir().join(PathBuf::from("faucet.toml"));
+        let faucet_account_path = temp_dir().join(PathBuf::from("account.mac"));
```

Comment:
From memory `join` takes `Into<Path>` or no?

```suggestion
        let config_path = temp_dir().join("faucet.toml");
        let faucet_account_path = temp_dir().join("account.mac");
```